### PR TITLE
Volto widgets view and extended DefaultView

### PR DESCRIPTION
**Pull request by ionlizarazu**
Related to #3259

### CODE REVIEW

File: src/helpers/Widget/utils.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export const getWidget = (id, field) => {
```
Comment (sneridagh) [inline src/helpers/Widget/utils.js:1]: Can we have a minimum test for this?

**Review by sneridagh — CHANGES_REQUESTED** [review]
LGTM! Could you please take a look at the comment?

File: src/components/theme/Widgets/BooleanWidget.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import cx from 'classnames';
+import { isBoolean } from 'lodash';
+
+const BooleanWidget = ({ value, children, className }) =>
+  isBoolean(value) ? (
+    <span className={cx(className, 'boolean', 'widget')}>
+      {children ? children(value ? 'Yes' : 'No') : value ? 'Yes' : 'No'}
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/BooleanWidget.jsx]: Done!

File: src/components/theme/Widgets/DateWidget.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import cx from 'classnames';
+import moment from 'moment';
+import config from '@plone/volto/registry';
+
+const DateWidget = ({ value, children, className }) => {
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/DateWidget.jsx]: Something like this?
https://github.com/plone/volto/blob/ae097ad5c269685adb79fa838958ad95d8dad451/src/components/theme/Widgets/DateWidget.jsx#L6

File: src/components/theme/Widgets/DatetimeWidget.jsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import cx from 'classnames';
+import moment from 'moment';
+import config from '@plone/volto/registry';
+
+const DatetimeWidget = ({ value, children, className }) => {
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/DatetimeWidget.jsx]: Something like this?

https://github.com/plone/volto/blob/ae097ad5c269685adb79fa838958ad95d8dad451/src/components/theme/Widgets/DatetimeWidget.jsx#L6

File: src/components/theme/Widgets/RelationWidget.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    <a
+      href={url}
+      className={cx(className, 'relation', 'widget', type, state)}
+      title={description}
+    >
+      <Icon
+        name={getContentIcon(type, value.is_folderish)}
+        size="20px"
+        className="icon-margin"
+        color="#878f93"
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/RelationWidget.jsx:27]: I've copied the code from ContentsItem.jsx component, feel free to modify it :smile: 

https://github.com/plone/volto/blob/f10fd036f19b80b70b90869e1240304d9ea5c921/src/components/manage/Contents/ContentsItem.jsx#L152

File: src/components/theme/Widgets/DatetimeWidget.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (giuliaghisini) [reply src/components/theme/Widgets/DatetimeWidget.jsx]: yess!

File: src/components/theme/Widgets/RelationWidget.jsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (giuliaghisini) [reply src/components/theme/Widgets/RelationWidget.jsx:27]: define styles in css and use variables!

[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/RelationWidget.jsx:27]: Maybe this is a work for another PR?

File: src/components/theme/Widgets/UrlWidget.jsx [hunk 6]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import cx from 'classnames';
+
+const UrlWidget = ({ value, children, className }) =>
+  value ? (
+    <a
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/UrlWidget.jsx]: this shoud work:

https://github.com/plone/volto/blob/21529a642e771ef62dbda58a5d496cc6ca37dfd7/src/components/theme/Widgets/UrlWidget.jsx#L7

File: src/components/theme/View/DefaultView.jsx [hunk 7]
```diff
@@ -32,8 +35,25 @@ const messages = defineMessages({
...
+    'settings',
+  ];
+  const fieldsets = contentSchema?.fieldsets.filter(
+    (fs) => !fieldsetsToExclude.includes(fs.id),
+  );
+
+  React.useEffect(() => {
+    dispatch(getSchema(content['@type'], location.pathname));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/View/DefaultView.jsx]: Done!

[the remark this answers is not in the document]
Comment (giuliaghisini) [reply src/components/theme/View/DefaultView.jsx]: i cannot se differences.. maybe not committed?

[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/View/DefaultView.jsx]: wrong link, sorry:

https://github.com/plone/volto/blob/543471fbeb70001dd1fb37321c3d0b722f99c241/src/components/theme/View/DefaultView.jsx#L54

File: src/components/theme/Widgets/UrlWidget.jsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (ionlizarazu) [reply src/components/theme/Widgets/UrlWidget.jsx]: It makes sense. done!

File: src/components/theme/Widgets/TokenWidget.jsx [hunk 8]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import cx from 'classnames';
+
+const TokenWidget = ({ value, children, className }) =>
+  value ? (
+    <span className={cx(className, 'token', 'widget')}>
+      {value.map((tag) => (
+        <a className="ui label" href={`/search?Subject=${tag}`} key={tag}>
```
Comment (giuliaghisini) [inline src/components/theme/Widgets/TokenWidget.jsx:8]: here too.. use UniversalLink where you do links

**Review by giuliaghisini — CHANGES_REQUESTED** [review]

File: src/components/theme/Widgets/TokenWidget.jsx (hunk 8 above)
Comment (ionlizarazu) [reply src/components/theme/Widgets/TokenWidget.jsx]: done:

https://github.com/plone/volto/commit/b83586d074fb501d1773a2271b2a0570dd8446c3

**Comment by sneridagh** [comment]
This is the current output:

image

Can we polish it a bit? I'd remove the left margin at least. What do you think?

Also, if you want to make @tiberiuichim happy, it would be great to have a StoryBook story for each :)

**Comment by ionlizarazu** [comment]
> This is the current output:
> 
> image
> 
> Can we polish it a bit? I'd remove the left margin at least. What do you think?
> 
> Also, if you want to make @tiberiuichim happy, it would be great to have a StoryBook story for each :)

I've removed the Segment tag to each field. Is it better like this?

image


Stories done.

**Review by sneridagh — APPROVED** [review]

**Comment by sneridagh** [comment]
@ionlizarazu AWESOME! Thanks a lot!

@giuliaghisini all good to merge now?

**Review by giuliaghisini — APPROVED** [review]

**Pull request merged**